routes/stripe.js

```diff
@@ -8,17 +8,30 @@ var stripe = require('stripe')(stripeKeys.secretKey);
 
 module.exports = {
   single: function(transaction, callback) {
-    var charge = {
-      amount: transaction.amount,
-      currency: transaction.currency,
-      card: transaction.stripeToken
-    };
-    stripe.charges.create(charge, callback);
+    stripe.customers.create({
+      email: transaction.email,
+      metadata: transaction.metadata,
+      source: transaction.stripeToken
+    }, function(err, customer) {
+      var charge = {};
+      if (err) {
+        callback(err);
+      } else {
+        charge = {
+          amount: transaction.amount,
+          currency: transaction.currency,
+          customer: customer.id,
+          description: transaction.description
+        };
+        stripe.charges.create(charge, callback);
+      }
+    });
   },
   recurring: function(transaction, callback) {
     stripe.customers.create({
       email: transaction.email,
-      metadata: transaction.metadata
+      metadata: transaction.metadata,
+      source: transaction.stripeToken
```

Comment:
Moving source (the user's credit card number) from subscription to customer. Wasn't a huge reason, but had to go somewhere and something about capturing it sooner rather than on a second request felt marginally better.